### DISCUSSION TITLE
Simple scale to margins

### DISCUSSION THREAD

**Discussion by marcrleonard** [abey79/vpype#665] [General]
Hello, I'm having a hell of time accomplishing something (seemingly) simple. Whenever I create art, I do so with the paper dimensions in mind. So if I'm plotting to an 11x14, I will set my width/height to 356x276mm. The vast majority of the time when I am preparing the the SVG for plotting all I want to do scale the entire thing to provide a margin around the outside (say, 1cm). With vpype, I can't seem to accomplish this in a simple way. I've tried:
`vpype read ./source.svg layout --fit-to-margins 1cm tight write --page-size 356x276mm ./output.svg` (source.svg included in this post)

![source](https://github.com/abey79/vpype/assets/24658895/d5741175-47d2-4ea8-a8e8-f5babae3a4e6)

This does provide the margin, but it completely changes where the artwork is. 

Unless I have a vast misunderstanding on how some of the commands work (which, I may), my overall impression is that there should be a a lot more inheritance of the original artwork so there is more of a 'do no harm' attitude. Things like moving/centering/write dimensions should all be a little less destructive. For instance, when calling `write someart.svg` it should write the output as the same dimensions as the source.

Anyways, just some thoughts! I think the tool is really great, as it provides all the buttons and switches you'd need. But for some reason, simple operations seem very hard to accomplish without a lot of commands.

**Comment by abey79** [comment]
Thanks for the feedback. From your description, I'm not 100% sure about the exact outcome you are looking for, in particular when it comes to the "out of bounds" part of your example SVG. Is that the expected output?

<img width="719" alt="image" src="https://github.com/abey79/vpype/assets/49431240/7b8d6d06-613f-4549-af17-a0aab4c0bd3a">

FWIW, this is how I obtained this result:
```
vpype read --no-crop input.svg layout -m 1cm --landscape 11x14in show
```

In any case, here are some relevant info on how *vpype* generally behaves:
- By default, `read` crops the geometries to the SVG boundaries. In your case, it means that 3/4th of your circle is cropped away. If that's not the intended behaviour, pass the `-c/--no-crop` option.
- The `write` command *does* honour the input SVG's bounds (or any bound set by other commands), *unless* you pass the `--page-size` argument, in which case those new bounds are used.
- The `--page-size`, `--center`, and `--landscape` options of the `write` command dates back from before `layout` existed and should generally *not* be used. I keep them around for backward compatibility reason, but they probably cause more harm than good.
- The `layout` command is the intended way to achieve what you seem to be wanting. It will do two things: (1) sets the page size to the provided argument and (2) scale/translate stuff to fit within the provided margin, with some control on vertical/horizontal alignment. As such, passing `tight` as argument to `layout` is typically not the intended behaviour.
- There are uses case that are not yet fully addressed by the `layout` command, e.g. see #663.
